homeassistant/components/zwave_js/fan.py

```diff
@@ -224,3 +235,110 @@ def zwave_speed_to_percentage(self, zwave_speed: int) -> int:
         # the UI handles steps e.g., for a 3-speed fan, you get steps at 33,
         # 67, and 100.
         return round(percentage)
+
+
+class ZwaveThermostatFan(ZWaveBaseEntity, FanEntity):
+    """Representation of a Z-Wave thermostat fan."""
+
+    _fan_mode: ZwaveValue
+    _fan_off: ZwaveValue | None = None
+    _fan_state: ZwaveValue | None = None
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize the thermostat fan."""
+        super().__init__(config_entry, client, info)
+
+        self._fan_mode = self.info.primary_value
+
+        self._fan_off = self.get_zwave_value(
+            THERMOSTAT_FAN_OFF_PROPERTY,
+            CommandClass.THERMOSTAT_FAN_MODE,
+            add_to_watched_value_ids=True,
+        )
+        self._fan_state = self.get_zwave_value(
+            THERMOSTAT_FAN_STATE_PROPERTY,
+            CommandClass.THERMOSTAT_FAN_STATE,
+            add_to_watched_value_ids=True,
+        )
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the device on."""
+        if self._fan_off:
+            await self.info.node.async_set_value(self._fan_off, False)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        if self._fan_off:
+            await self.info.node.async_set_value(self._fan_off, True)
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if device is on."""
+        if (value := get_value_of_zwave_value(self._fan_off)) is None:
+            return None
+        return not cast(bool, value)
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite."""
+        value = get_value_of_zwave_value(self._fan_mode)
+        if value is None or str(value) not in self._fan_mode.metadata.states:
+            return None
+        return cast(str, self._fan_mode.metadata.states[str(value)])
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+
+        try:
+            new_state = int(
+                next(
+                    state
```

Comment:
We could put the copy to integer here instead around the state so we don't need to indent as much.